wagtail/contrib/forms/views.py

```diff
@@ -215,7 +203,7 @@ def get_validated_ordering(self):
             #  Revert to CSV order_by submit_time ascending for backwards compatibility
             default_ordering = self.ordering_csv or ()
         else:
-            default_ordering = self.ordering or ()
+            default_ordering = self.default_ordering or ()
```

Comment:
I opted not to change the Python code too much because we documented this view to be customisable. You'll see that this view class still has its own stuff like:
- `get_ordering` with its bespoke `get_validated_ordering`
- using some bespoke Python logic for the data, and rendering the table without the tables UI framework

It's a bit risky to fully embrace the generic views approach here and I don't think it's worth the time, effort, and breaking changes (at least for now). Happy to revisit later.

Or if you think we might as well do the total refactoring (with tables framework etc) in one go alongside the changes in this PR (so that developers only need to make adjustments once), I guess I can try to do that.